### PR TITLE
[Core] Fix in-container memory limit fetching for cgroups v2

### PR DESCRIPTION
**Pull request by clarkzinzow**
When trying to fetch the in-container memory limit while running Ray in a container using cgroups v2, the limit parsing will fail if the limit is not set, which will prevent Ray from being able to start. This currently happens when trying to run Ray in a Google Colab notebook.

It looks like this memory limit fetching path might be untested? cc @DmitriGekhtman @wuisawesome 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/utils.py [hunk 1]
```diff
@@ -389,7 +389,12 @@ def get_system_memory():
             docker_limit = int(f.read())
     elif os.path.exists(memory_limit_filename_v2):
         with open(memory_limit_filename_v2, "r") as f:
-            docker_limit = int(f.read())
+            max_file = f.read()
+            if max_file.isnumeric():
```
Comment (ericl) [inline python/ray/_private/utils.py:393]: Huh never know this existed.

**Review by ericl — APPROVED** [review]

**Review by DmitriGekhtman — DISMISSED** [review]
Thanks for the fix!
Looks like I neglected to test memory in https://github.com/ray-project/ray/pull/21688 and only covered CPU.
Would you mind adding a test for the fix in this PR? test_advanced_3 is as good a place as any to put it.

**Comment by richardliaw** [comment]
LGTM, adding test would be great.

**Comment by clarkzinzow** [comment]
@DmitriGekhtman @richardliaw Tests added.

**Comment by richardliaw** [comment]
seems like test failures are unrelated, merging

**Pull request merged**